src/ML-IAP/mliap_model_python.cpp

```diff
@@ -66,17 +70,18 @@ MLIAPModelPython::MLIAPModelPython(LAMMPS *lmp, char *coefffilename) :
     PyList_Append(py_path, PY_STRING_FROM_STRING(potentials_path));
   }
   PyGILState_Release(gstate);
-
   if (coefffilename) read_coeffs(coefffilename);
 
-  nonlinearflag = 1;
+
 }
 
 /* ---------------------------------------------------------------------- */
 
 MLIAPModelPython::~MLIAPModelPython()
 {
-  MLIAPPY_unload_model(this);
+  if (model_loaded)
+    MLIAPPY_unload_model(this);
+  model_loaded=false;
```

Comment:
model_loaded is an integer variable not a boolean.

Reply:
fixed